torch_xla/csrc/tensor_methods.cpp

```diff
@@ -468,6 +470,50 @@ XLATensor XLATensor::_adaptive_avg_pool2d_backward(const XLATensor& grad_output,
       grad_output.GetIrValue(), input.GetIrValue()));
 }
 
+void XLATensor::_amp_foreach_non_finite_check_and_unscale_(
+    std::vector<XLATensor> self, XLATensor& found_inf,
+    const XLATensor& inv_scale) {
+  std::vector<ir::Value> inputs;
+  for (const auto& x : self) {
+    inputs.push_back(x.GetIrValue());
+  }
+  inputs.push_back(found_inf.GetIrValue());
+  inputs.push_back(inv_scale.GetIrValue());
```

Comment:
I think it is better to leave `found_inf ` and `inv_scale ` to be separate arguments instead of adding them to input lists since they have different meanings. You can refer to `XLATensor::all_reduce` in  the same file, which also takes a `std::vector<XLATensor>`.

This will also make output shape computation a bit less puzzling, `inputs.size() - 2` was a bit confusing to read.

Reply:
Done.